src/ZoneServer/Scripting/Shortcuts.cs

```diff
@@ -154,109 +155,104 @@ public static WarpMonster AddWarp(string warpName, double direction, Location fr
+		/// initial spawn points
 		/// </summary>
 		/// <param name="monsterClassId"></param>
 		/// <param name="maxAmount"></param>
 		/// <param name="respawn"></param>
-		/// <param name="map"></param>
-		/// <param name="area"></param>
+		/// <param name="spawnerName"></param>
+		/// <param name="minAmount"></param>
 		/// <param name="tendency"></param>
 		/// <param name="propertyOverrides"></param>
 		/// <returns></returns>
-		/// <exception cref="ArgumentException"></exception>
-		public static MonsterSpawner AddSpawner(int monsterClassId, int maxAmount, TimeSpan respawn, string map, IShape area, TendencyType tendency, PropertyOverrides propertyOverrides)
+		public static MonsterSpawner AddSpawner(int monsterClassId, int maxAmount, TimeSpan respawn, string spawnerName = "", int minAmount = -1, TendencyType tendency = TendencyType.Peaceful, PropertyOverrides propertyOverrides = null)
 		{
 			var initialSpawnDelay = TimeSpan.FromSeconds(0);
 			var minRespawnDelay = Math2.Max(TimeSpan.FromSeconds(3), respawn);
 			var maxRespawnDelay = minRespawnDelay.Multiply(3);
-			var minAmount = Math.Max(1, (int)(maxAmount * 0.05));
+			var maxSpawnAmount = Math.Max(1, maxAmount);
+
+			// Creates spawner
+			var spawner = AddSpawner(monsterClassId, maxSpawnAmount, initialSpawnDelay, minRespawnDelay, maxRespawnDelay, spawnerName, minAmount, tendency, propertyOverrides);
 
-			return AddSpawner(monsterClassId, minAmount, maxAmount, initialSpawnDelay, minRespawnDelay, maxRespawnDelay, map, area, tendency, propertyOverrides);
+			return spawner;
 		}
 
 		/// <summary>
-		/// Adds monster spawner to the world.
+		/// Base shortcut for adding spawner
 		/// </summary>
+		/// <param name="spawnerName"></param>
 		/// <param name="monsterClassId"></param>
 		/// <param name="minAmount"></param>
 		/// <param name="maxAmount"></param>
 		/// <param name="initialSpawnDelay"></param>
 		/// <param name="minRespawnDelay"></param>
 		/// <param name="maxRespawnDelay"></param>
-		/// <param name="map"></param>
-		/// <param name="area"></param>
 		/// <param name="tendency"></param>
 		/// <param name="propertyOverrides"></param>
 		/// <returns></returns>
 		/// <exception cref="ArgumentException"></exception>
-		public static MonsterSpawner AddSpawner(int monsterClassId, int minAmount, int maxAmount, TimeSpan initialSpawnDelay, TimeSpan minRespawnDelay, TimeSpan maxRespawnDelay, string map, IShape area, TendencyType tendency, PropertyOverrides propertyOverrides)
+		public static MonsterSpawner AddSpawner(int monsterClassId, int maxAmount, TimeSpan initialSpawnDelay, TimeSpan minRespawnDelay, TimeSpan maxRespawnDelay, string spawnerName = "", int minAmount = -1, TendencyType tendency = TendencyType.Peaceful, PropertyOverrides propertyOverrides = null)
 		{
-			if (!ZoneServer.Instance.World.TryGetMap(map, out var mapObj))
-				throw new ArgumentException($"Map '{map}' not found.");
+			if (spawnerName.IsNullOrWhiteSpace())
```

Comment:
If you just create the spawner but don't add spawn points to it it won't be able to spawn anything, yes.

We opted for this architecture where spawners are not necessarily associated to a map but the spawn points are. We can't have spawners also associated to the map because then we can't do cross-map spawning (required for world bosses).

It still makes sense, however, to create a spawner with an empty name string (which will be internally generated) because that shortcut returns the object and the "Name" property is public, so they can always find what name the spawner was generated with.